File: src/main/java/org/hildan/bots/riseoflords/util/Format.java
package org.hildan.bots.riseoflords.util;

import java.text.DecimalFormat;
import java.text.DecimalFormatSymbols;

public class Format {

    private static final DecimalFormat fmt;
    static {
        final DecimalFormatSymbols symbols = DecimalFormatSymbols.getInstance();
        symbols.setGroupingSeparator('.');
        fmt = new DecimalFormat("###,###.##", symbols);
    }

    public static String gold(int amount) {
        return fmt.format(amount);
    }

}

File: src/main/kotlin/org/hildan/bots/riseoflords/util/Format.kt
package org.hildan.bots.riseoflords.util

import java.text.DecimalFormat
import java.text.DecimalFormatSymbols

object Format {
    private val symbols = DecimalFormatSymbols.getInstance().apply { groupingSeparator = '.' }
    private val fmt: DecimalFormat = DecimalFormat("###,###.##", symbols)

    fun gold(amount: Int): String = fmt.format(amount.toLong())
}